packages/teleterm/src/ui/DocumentTerminal/DocumentTerminal.tsx

```diff
@@ -14,12 +14,50 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import React from 'react';
+import React, { useEffect } from 'react';
 import Document from 'teleterm/ui/Document';
 import useDocTerminal, { Props } from './useDocumentTerminal';
 import Terminal from './Terminal';
+import { useAppContext } from 'teleterm/ui/appContextProvider';
+import { ResourceReconnect } from 'teleterm/ui/ResourceReconnect';
 
-export default function DocumentTerminal(props: Props & { visible: boolean }) {
+export default function DocumentTerminalContainer(props: Props) {
+  const ctx = useAppContext();
+  const { doc } = props;
+
+  useEffect(() => {
+    if (
+      doc.kind === 'doc.terminal_tsh_node' &&
+      ctx.serviceClusters.findCluster(`/clusters/${doc.rootClusterId}`)
```

Comment:
I'm not sure about it, is there any better way to find cluster by id?

Reply:
I would try to replace  `login`, `rootClusterId`, and `leafClusterId` with `serverUri` (or just add serverUri to existing props) then you can do `ctx.clustersService.findClusterByResource(serverUri)` to find the cluster.

Reply:
Good idea, thanks :) 
I won't be removing these properties for now, they seem useful.